gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ListMapper.java

```diff
@@ -0,0 +1,155 @@
+        ListResource topic = new ListResource();
+
+        topic
+            .setStatus(ListStatus.CURRENT)
+            .setMode(ListMode.SNAPSHOT)
+            .setTitle(getTitle(compoundStatement))
+            .setCode(getListCoding(LIST_CODE_SYSTEM, TOPIC_CODE_CODE, TOPIC_CODE_DISPLAY))
+            .setEncounter(consultation.getEncounter())
+            .setSubject(consultation.getSubject())
+            .setDateElement(getDate(compoundStatement, consultation))
+            .setOrderedBy(getListCoding(LIST_ORDERED_BY_SYSTEM, LIST_ORDERED_BY_CODE, LIST_ORDERED_BY_DISPLAY))
+            .setMeta(getListMeta())
+            .setId(getTopicId(compoundStatement));
+
+        return topic;
+    }
+
+    private String getTopicId(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return compoundStatement != null ? compoundStatement.getId().get(0).getRoot() : idGenerator.generateUuid(); // TODO: double check
+    }
+
+    public ListResource mapToCategory(ListResource topic, RCMRMT030101UK04CompoundStatement compoundStatement) {
+        ListResource category = new ListResource();
+
+        category
+            .setStatus(ListStatus.CURRENT)
+            .setMode(ListMode.SNAPSHOT)
+            .setTitle(getTitle(compoundStatement))
+            .setCode(getListCoding(LIST_CODE_SYSTEM, CATEGORY_CODE_CODE, CATEGORY_CODE_DISPLAY))
+            .setEncounter(topic.getEncounter())
+            .setSubject(topic.getSubject())
+            .setDateElement(getDate(compoundStatement, topic))
+            .setOrderedBy(getListCoding(LIST_ORDERED_BY_SYSTEM, LIST_ORDERED_BY_CODE, LIST_ORDERED_BY_DISPLAY))
+            .setMeta(getListMeta())
+            .setId(compoundStatement.getId().get(0).getRoot());
+
+        return category;
+    }
+
+    private DateTimeType getDate(RCMRMT030101UK04CompoundStatement compoundStatement, ListResource parentList) {
+        if (compoundStatement != null && compoundStatement.getAvailabilityTime().getValue() != null) {
+            return DateFormatUtil.parseToDateTimeType(compoundStatement.getAvailabilityTime().getValue());
+        }
+
+        return parentList.getDateElement();
+    }
+
+    private String getTitle(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        if (compoundStatement != null) {
+            var codeableConcept = codeableConceptMapper.mapToCodeableConcept(compoundStatement.getCode());
+            if (codeableConcept.hasText()) {
+                return codeableConcept.getText();
+            } else if (codeableConcept.getCoding().get(0).hasDisplay()) {
+                return codeableConcept.getCoding().get(0).getDisplay();
+            }
+        }
+
+        return null;
+    }
+
+    private Meta getListMeta() {
+        return new Meta().addProfile(LIST_META_PROFILE);
+    }
+
+    private CodeableConcept getListCoding(String system, String code, String display) {
```

Comment:
I'm not sure if this could be useful, but there is a `CodeableConceptUtils` class, if you'd like you can move it to the common module and use here.